test/LiquidityManager/invest.test.ts

```diff
@@ -82,12 +113,15 @@ describe.only('LiquidityManager#invest', () => {
         return addressTokenA.toLowerCase() === addressTokenB.toLowerCase()
     }
 
-    function getEncodedSwap(
+    async function getEncodedSwap(
         amount: bigint,
         outputToken: AddressLike,
         outputTokenPrice: BigNumber,
         protocol: 'uniswapV2' | 'uniswapV3' = 'uniswapV2',
     ) {
+        if (!amount || await isSameToken(stablecoin, outputToken))
```

Comment:
what if we move this logic to the encodeSwap functions? also how and why would the amount be 0?

Reply:
the amount can be 0 when we have the price ranges outside the current price, in this case amount0 or amount1 will be 0